src/utils/DepositUtils.ts

```diff
@@ -71,3 +73,38 @@ export async function queryHistoricalDepositForFill(
 
   return validateFillForDeposit(fill, deposit) ? deposit : undefined;
 }
+
+export function createDepositForSimulatingGas(
+  amountToRelay: BigNumberish,
+  tokenSymbol: string,
+  originChainId: string,
+  destinationChainId: string,
+  relayerAddress: string,
+  messagePayload?: {
+    message: string;
+    recipientAddress: string;
+  }
+): Deposit {
+  const originToken = resolveContractFromSymbol(tokenSymbol, originChainId);
+  const destinationToken = resolveContractFromSymbol(tokenSymbol, destinationChainId);
+  if (!isDefined(originToken) || !isDefined(destinationToken)) {
+    throw new Error(`Could not resolve token contract for ${tokenSymbol} on ${originChainId} or ${destinationChainId}`);
+  }
+
+  return {
+    // We want to set the deposit ID to 0 so that there
+    // is no chance of a collision with a real deposit
+    depositId: 0,
+    amount: toBN(amountToRelay),
+    depositor: relayerAddress ?? ZERO_ADDRESS,
```

Comment:
nit: prefer a known default (like we do today for recipient), random address, or [0xFFFFF....](https://etherscan.io/address/0xffffffffffffffffffffffffffffffffffffffff), since the zero address will probably underestimate the calldata costs (0 bytes are cheaper than nonzero bytes).

Reply:
Fair point - I've made the depositorAddress default to a random address to prevent overestimating with 0xFFFF... unless overestimating would be beneficial.

https://github.com/across-protocol/sdk-v2/pull/427/commits/c08699837a7c78648734ff969f0a74fa275ef890

Reply:
Done: https://github.com/across-protocol/sdk-v2/pull/427/commits/c08699837a7c78648734ff969f0a74fa275ef890

Reply:
Overestimating can have a bad edge case where relayer refuses to fill a deposit because its too expensive, causing innocent user deposit to be ignored. I'd err on the side of the relayer losing money versus the user